src/widgets/AddMainTask/AddMainTask.tsx

```diff
@@ -0,0 +1,99 @@
+import { ChangeEvent, memo, useState } from 'react';
+import {
+    Box, Button, Checkbox, FormControlLabel, TextField,
+} from '@mui/material';
+import { createTask, createTaskDto } from 'entities/Task';
+import { useAppDispatch, useAppSelector } from 'shared/lib/hooks/redux';
+import { getUserData } from 'entities/User';
+import { useTranslation } from 'react-i18next';
+import { FormValidator } from 'shared/lib/helpers';
+
+import { AddMainTaskStyle as styles } from './AddMainTask.style';
+
+export const AddMainTask = memo(() => {
```

Comment:
part of this component already exist as `AddTask`, seems the code can be reused instead of duplicating

Reply:
Done, but had to add some corrections that I'm not happy with, but without them, the function doesn't work the same way as when there were separate components. 
If you have any suggestions on how to do it more correctly, I would be glad to know.